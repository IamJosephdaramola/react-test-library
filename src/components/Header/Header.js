import { Fragment } from "react";
import "./Header.css";

export default function Header({ title }) {
  return (
    <Fragment>
      <h1 className="header" data-testid="header-1">
        {title}
      </h1>
      {/* <h3 title='Header' className="header">Cats</h3> */}
    </Fragment>
  );
}
