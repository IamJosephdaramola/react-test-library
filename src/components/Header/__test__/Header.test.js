import { render, screen, cleanup } from "@testing-library/react";
import Header from "../Header";

afterEach(cleanup);

test("should take a snapshot of the header", () => {
  const { asFragment } = render(<Header title="todo list" />);
  expect(asFragment(<Header />)).toMatchSnapshot();
});

// Get by
test("should render header by test correctly", () => {
  render(<Header title="todo list" />);

  const HeaderElement = screen.getByText(/todo list/i);
  expect(HeaderElement).toBeInTheDocument();
});

/*
test('should render header by role correctly', () => {
	render(<Header title='todo list' />);

	const HeaderElement = screen.getByRole('heading', {
		name: 'todo list',
	});
	expect(HeaderElement).toBeInTheDocument();
});

test('should render header by title correctly', () => {
	render(<Header title='todo list' />);

	const HeaderElement = screen.getByTitle('Header');
	expect(HeaderElement).toBeInTheDocument();
});

test('should render header by id correctly', () => {
	render(<Header title='todo list' />);

	const HeaderElement = screen.getByTestId('header-1');
	expect(HeaderElement).toBeInTheDocument();
});

// Find by
test('should render header by test (find) correctly', async () => {
	render(<Header title='todo list' />);

	const HeaderElement = await screen.findByText(/todo list/i);
	expect(HeaderElement).toBeInTheDocument();
});

// Query by
test('should render header by test (query) correctly', () => {
	render(<Header title='todo list' />);

	const HeaderElement = screen.queryByText(/todo st/i);
	expect(HeaderElement).not.toBeInTheDocument();
});

// Get all by
test('should render header by test (get all) correctly', () => {
	render(<Header title='todo list' />);

	const HeaderElements = screen.getAllByRole('heading');
	expect(HeaderElements.length).toBe(2);
});
*/
