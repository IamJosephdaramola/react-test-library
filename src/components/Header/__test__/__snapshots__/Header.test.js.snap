// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of the header 1`] = `
<DocumentFragment>
  <h1
    class="header"
    data-testid="header-1"
  >
    todo list
  </h1>
</DocumentFragment>
`;
